src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -196,6 +208,46 @@ public void ConnectHeaders_SupplyHeadersThenSupplyMore_Both_Tip_PeerId_Maps_Shou
             Assert.True(peerTipsByPeerIdBefore.Values.Count == peerTipsByPeerIdAfter.Values.Count);
         }
 
+        /// <summary>
```

Comment:
I don't think this is a good idea to use Summaries here to describe the scenario of the test.

This can be embedded into the test itself with better naming of the method (so that it shows in test runners like resharper)  but also in the code itself with method names.

Disconnecting the scenario from the code means that any developer needing to change this in the future has to read the scenario and try to correlate it with the blocks of code and this cognitive load is very high.

If we are writing tests just to sit there as regression and never come back to them then sure it just doesn;t matter - since we know that software almost always changes then lets make the blocks mean something.

Reply:
Some of scenarios are too complex to put in a test name, so I'd keep the desriptions there.

Also it helps during the review to compare the intention vs implementation.

Reply:
Test Names should represent the high level - that's fine. The scenario itself can be documented inside with meaningful method names and or nicer test domain model / extensions - this would also help with review,

Reply:
A compromise here might be to allow comments during test development and then a cleanup at the end of task.